tests/semantic/test_semantics.py

```diff
@@ -0,0 +1,186 @@
+import asyncio
+import json
+import os
+import re
+from pathlib import Path
+
+import pytest
+import pytest_check as check
+from starkware.cairo.lang.vm.vm_exceptions import VmException
+from starkware.starknet.compiler.compile import compile_starknet_files
+from starkware.starknet.testing.state import StarknetState
+from starkware.starkware_utils.error_handling import StarkException
+
+from warp.yul.main import transpile_from_solidity
+from warp.yul.starknet_utils import (
+    deploy_contract,
+    deploy_contract_evm_calldata,
+    invoke_method,
+    invoke_method_evm_calldata,
+)
+
+semantic_tests_folder = os.path.dirname(__file__)
+warp_root = os.path.join(semantic_tests_folder, "../../")
+test_folder = os.path.join(
+    semantic_tests_folder, "solidity/test/libsolidity/semanticTests"
+)
+
+CONTRACT_DEF_REGEX = re.compile(r"contract\s+(\w+)\s+")
+
+
+def get_last_contract(contract_file):
+    contract_name = None
+    with open(contract_file, "r") as contract:
+        lines = contract.readlines()
+        for line in lines:
+            matches = CONTRACT_DEF_REGEX.match(line)
+            if matches:
+                contract_name = matches.groups()[0]
+
+    if contract_name:
+        return contract_name
+
+    pytest.skip(f"Could not find any contract definitions in {contract_file}")
+
+
+def load_ignore_list():
+    with open(
+        os.path.abspath(os.path.join(semantic_tests_folder, "ignore_files.json"))
+    ) as ignore_list:
+        data = json.load(ignore_list)
+
+        if data:
+            return data["files"]
+
+    pytest.skip("Ignore list not found")
+
+
+def load_test_calldata():
+    with open(
+        os.path.abspath(os.path.join(semantic_tests_folder, "test_calldata.json"))
+    ) as calldata:
+        return json.load(calldata)
+
+    pytest.skip("calldata for tests could not be loaded")
+
+
+ignore_list = load_ignore_list()
+test_calldata = load_test_calldata()
+
+
+async def starknet_compile(contract_file):
+    cairo_path = f"{warp_root}/src/warp/cairo-src"
+    return compile_starknet_files(
+        [contract_file], debug_info=True, cairo_path=[cairo_path]
+    )
+
+
+def sol_files_in_dir(folder):
+    # calling str on PosixPath, so that when the test fails it shows the file
+    # name instead of an object
+    return [str(f) for f in Path(folder).glob("**/*.sol")]
```

Comment:
But printing a path still prints just the file name?.. 
![image](https://user-images.githubusercontent.com/44875844/150112594-b21d5cc4-c180-4b0e-aee7-7c7e8d69d2cb.png)

Reply:
It does not have the same behaviour when the paths are pytest parameterized, they get printed as `contract_file<n>`

Reply:
Oh, weird. I see